transport/x/grpc/inbound.go

```diff
@@ -85,22 +91,17 @@ func (i *Inbound) start() error {
 	if i.router == nil {
 		return errRouterNotSet
 	}
-	serviceDescs, err := i.getServiceDescs()
-	if err != nil {
-		return err
-	}
+
+	i.handler = newHandler(
```

Comment:
If there's no reason to create state, then it's best not to create state at all, ie this is a static variable (class variable depending on your language), not an instance variable - always default to creating less state, not more. Ask that you just do `handler := newHandler(...)`, can you do that?